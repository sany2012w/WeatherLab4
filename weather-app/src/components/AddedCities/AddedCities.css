.weather {
  display: grid;
  grid-template-columns: 1fr 1fr;
  grid-gap: 30px;
  margin: 20px 0;
}
