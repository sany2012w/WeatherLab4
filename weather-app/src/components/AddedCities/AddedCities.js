import React from "react";
import { connect } from "react-redux";
import AddPanel from '../AddPanel/AddPanel';
import WeatherData from '../WeatherData/WeatherData';
import { fetchWeatherByCityName } from '../../actions/fetchWeatherByCityName';
import { getAddedCities, deleteCity, addCity } from '../../actions/fetchToServerFavorites';
import './AddedCities.css';


class AddedCities extends React.Component {
  componentDidMount() {
    this.props.getAddedCities();
  }

  render() {
    return (
      <div>
        <AddPanel onSubmit={(e) => this.addNewCity(e)} />
        {this.props.error && <div className="error">Error: {this.props.error}</div>}
        <div className="weather">
          {
            [...this.props.added_cities.entries()].map((entry) => {
              return (
                <WeatherData
                  key={entry[0]}
                  onFetch={() => this.props.fetchWeatherByCityName(entry[0])}
                  onDelete={() => this.props.deleteCity(entry[0], false)}
                  data={entry[1]}/>
              );
            })
          }
        </div>
      </div>
    );
  }

  addNewCity(e) {
    e.preventDefault();
    this.props.addCity(e.currentTarget.elements.cityName.value);
  }
}




function mapStateToProps(state) {
  return {
    added_cities: state.cities.added_cities,
    error: state.cities.error
  };
}

function mapDispatchToProps(dispatch) {
  return {
    getAddedCities: () => {
      dispatch(getAddedCities());
    },

    addCity: (cityName) => {
      dispatch(addCity(cityName));
    },

    deleteCity: (cityName) => {
      dispatch(deleteCity(cityName));
    },

    fetchWeatherByCityName: (cityName) => {
      dispatch(fetchWeatherByCityName(cityName));
    }
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(AddedCities);
