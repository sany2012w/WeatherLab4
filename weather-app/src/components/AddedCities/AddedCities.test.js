import React from "react";
import AddedCities from "./AddedCities";
import renderer from "react-test-renderer";
import weatherTestData from "./weatherTestData";
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux'
import thunk from "redux-thunk";

const testStore = configureMockStore([thunk]);

describe("AddedCities are displayed correctly", () => {
    it("There are cities in the favorites", () => {
      let testCities = new Map([["London", weatherTestData], ["London1", weatherTestData]]);
      const store = testStore({
        cities: {
              added_cities: testCities
          }
        });

      const tree = renderer.create(
        <Provider store={store}>
          <AddedCities/>
       </Provider>).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it("There are no cities in the favorites", () => {
      let testCities = new Map();
      const store = testStore({
        cities: {
              added_cities: testCities
          }
        });
      const tree = renderer.create(
        <Provider store={store}>
          <AddedCities/>
       </Provider>).toJSON();
      expect(tree).toMatchSnapshot();
    });
})


describe("Error", () => {
    it("There are cities in the favorites", () => {
      let testCities = new Map([["London", weatherTestData], ["London1", weatherTestData]]);
      const store = testStore({
        cities: {
              added_cities: testCities,
              error: 'Something wrong'
          }
        });

      const tree = renderer.create(
        <Provider store={store}>
          <AddedCities/>
       </Provider>).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it("There are no cities in the favorites", () => {
      let testCities = new Map();
      const store = testStore({
        cities: {
              added_cities: testCities,
              error: 'Something wrong'
          }
        });
      const tree = renderer.create(
        <Provider store={store}>
          <AddedCities/>
       </Provider>).toJSON();
      expect(tree).toMatchSnapshot();
    });
})
