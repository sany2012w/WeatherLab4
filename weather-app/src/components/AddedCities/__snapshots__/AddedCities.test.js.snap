// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddedCities are displayed correctly There are cities in the favorites 1`] = `
<div>
  <div
    className="CitiesPanel"
  >
    <div
      className="CitiesText"
    >
      Favorites
    </div>
    <form
      onSubmit={[Function]}
    >
      <input
        className="CitiesInput"
        name="cityName"
        placeholder="City name"
        required={true}
        type="text"
      />
      <input
        className="CitiesButton"
        type="submit"
        value="Add city"
      />
    </form>
  </div>
  <div
    className="weather"
  >
    <div
      className="WeatherBlock"
    >
      <div
        className="City"
      >
        London
      </div>
      <div
        className="leftWeather"
      >
        <img
          className="Icon"
          src="https://openweathermap.org/img/wn/09d.png"
        />
        <div
          className="Temp"
        >
          7
           째C
        </div>
        <button
          className="button"
          onClick={[Function]}
        >
          Delete
        </button>
      </div>
      <div
        className="rightWeather"
      >
        <div
          className="Pressure"
        >
          Pressure: 
          1012
           hPa
        </div>
        <div
          className="Humidity"
        >
          Humidity: 
          81
          %
        </div>
        <div
          className="Clouds"
        >
          Clouds: 
          light intensity drizzle
        </div>
        <div
          className="Wind"
        >
          Wind: 
          4.1
           m/s
        </div>
        <div
          className="Coords"
        >
          Coordinates: [
          -0.13
          , 
          51.51
          ]
        </div>
      </div>
    </div>
    <div
      className="WeatherBlock"
    >
      <div
        className="City"
      >
        London
      </div>
      <div
        className="leftWeather"
      >
        <img
          className="Icon"
          src="https://openweathermap.org/img/wn/09d.png"
        />
        <div
          className="Temp"
        >
          7
           째C
        </div>
        <button
          className="button"
          onClick={[Function]}
        >
          Delete
        </button>
      </div>
      <div
        className="rightWeather"
      >
        <div
          className="Pressure"
        >
          Pressure: 
          1012
           hPa
        </div>
        <div
          className="Humidity"
        >
          Humidity: 
          81
          %
        </div>
        <div
          className="Clouds"
        >
          Clouds: 
          light intensity drizzle
        </div>
        <div
          className="Wind"
        >
          Wind: 
          4.1
           m/s
        </div>
        <div
          className="Coords"
        >
          Coordinates: [
          -0.13
          , 
          51.51
          ]
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`AddedCities are displayed correctly There are no cities in the favorites 1`] = `
<div>
  <div
    className="CitiesPanel"
  >
    <div
      className="CitiesText"
    >
      Favorites
    </div>
    <form
      onSubmit={[Function]}
    >
      <input
        className="CitiesInput"
        name="cityName"
        placeholder="City name"
        required={true}
        type="text"
      />
      <input
        className="CitiesButton"
        type="submit"
        value="Add city"
      />
    </form>
  </div>
  <div
    className="weather"
  />
</div>
`;

exports[`Error There are cities in the favorites 1`] = `
<div>
  <div
    className="CitiesPanel"
  >
    <div
      className="CitiesText"
    >
      Favorites
    </div>
    <form
      onSubmit={[Function]}
    >
      <input
        className="CitiesInput"
        name="cityName"
        placeholder="City name"
        required={true}
        type="text"
      />
      <input
        className="CitiesButton"
        type="submit"
        value="Add city"
      />
    </form>
  </div>
  <div
    className="error"
  >
    Error: 
    Something wrong
  </div>
  <div
    className="weather"
  >
    <div
      className="WeatherBlock"
    >
      <div
        className="City"
      >
        London
      </div>
      <div
        className="leftWeather"
      >
        <img
          className="Icon"
          src="https://openweathermap.org/img/wn/09d.png"
        />
        <div
          className="Temp"
        >
          7
           째C
        </div>
        <button
          className="button"
          onClick={[Function]}
        >
          Delete
        </button>
      </div>
      <div
        className="rightWeather"
      >
        <div
          className="Pressure"
        >
          Pressure: 
          1012
           hPa
        </div>
        <div
          className="Humidity"
        >
          Humidity: 
          81
          %
        </div>
        <div
          className="Clouds"
        >
          Clouds: 
          light intensity drizzle
        </div>
        <div
          className="Wind"
        >
          Wind: 
          4.1
           m/s
        </div>
        <div
          className="Coords"
        >
          Coordinates: [
          -0.13
          , 
          51.51
          ]
        </div>
      </div>
    </div>
    <div
      className="WeatherBlock"
    >
      <div
        className="City"
      >
        London
      </div>
      <div
        className="leftWeather"
      >
        <img
          className="Icon"
          src="https://openweathermap.org/img/wn/09d.png"
        />
        <div
          className="Temp"
        >
          7
           째C
        </div>
        <button
          className="button"
          onClick={[Function]}
        >
          Delete
        </button>
      </div>
      <div
        className="rightWeather"
      >
        <div
          className="Pressure"
        >
          Pressure: 
          1012
           hPa
        </div>
        <div
          className="Humidity"
        >
          Humidity: 
          81
          %
        </div>
        <div
          className="Clouds"
        >
          Clouds: 
          light intensity drizzle
        </div>
        <div
          className="Wind"
        >
          Wind: 
          4.1
           m/s
        </div>
        <div
          className="Coords"
        >
          Coordinates: [
          -0.13
          , 
          51.51
          ]
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Error There are no cities in the favorites 1`] = `
<div>
  <div
    className="CitiesPanel"
  >
    <div
      className="CitiesText"
    >
      Favorites
    </div>
    <form
      onSubmit={[Function]}
    >
      <input
        className="CitiesInput"
        name="cityName"
        placeholder="City name"
        required={true}
        type="text"
      />
      <input
        className="CitiesButton"
        type="submit"
        value="Add city"
      />
    </form>
  </div>
  <div
    className="error"
  >
    Error: 
    Something wrong
  </div>
  <div
    className="weather"
  />
</div>
`;
